packages/dataviews/src/types.ts

```diff
@@ -482,11 +482,23 @@ export interface ViewTable extends ViewBase {
 		 * The density of the view.
 		 */
 		density?: Density;
+
+		/**
+		 * Whether infinite scroll is enabled for the table.
+		 */
+		infiniteScroll?: boolean;
 	};
 }
 
 export interface ViewList extends ViewBase {
 	type: 'list';
+
+	layout?: {
```

Comment:
So far, the `layout` object contained props that were specific to a particular layout and didn't make sense in the others, while infinite scroll seems to fall in the general view config category. Do you see it differently?

Reply:
I'm not sure. Currently it applies to all layouts, but will that always be the case? I don't have a strong opinion either way. Where would be a good place for a general setting, just in the view object?

Reply:
Or added directly to `ViewBase` rather than the individual ViewList, ViewTable, ViewGrid types?

I feel like that might be what @oandregal was referring to.

Reply:
You mean by adding a `layout` object to `ViewBase`? If layout is supposed to be for layout-specific things, I'm not sure we should do that.

Reply:
I mean directly at the root of `ViewBase`.

It is optional like most of the other properties so it can still be omitted from the config for particular layouts that don't support it.

Reply:
Ok to be clear, the type being part of `ViewBase` also means that the actual property goes directly in the view object? That's what I was asking initially. Types will naturally be updated as a consequence of that change.

Reply:
I've updated to move `infiniteScroll` out of `layout` and directly into `view. This has the side-effect of keeping the state of the toggle when switching between views (whereas before it reset).